tests/functional/filter/cram/filter-force-include-no-duplicates.t

```diff
@@ -0,0 +1,69 @@
+Setup
+
+  $ pushd "$TESTDIR" > /dev/null
+  $ source _setup.sh
+
+
+Test that a force-included strain is only output once.
+
+
+Create some files for testing.
+
+  $ cat >$TMP/metadata.tsv <<~~
+  > strain	col
+  > a	1
+  > b	2
+  > c	3
+  > d	4
+  > ~~
+  $ cat >$TMP/sequences.fasta <<~~
+  > >a
+  > NNNN
+  > >b
+  > NNNN
+  > >c
+  > NNNN
+  > >d
+  > NNNN
+  > ~~
+
+Test all outputs with --include-where.
+
+  $ ${AUGUR} filter \
+  >   --metadata $TMP/metadata.tsv \
+  >   --sequences $TMP/sequences.fasta \
+  >   --subsample-max-sequences 4 \
+  >   --include-where col=1 \
+  >   --subsample-seed 0 \
+  >   --output-metadata $TMP/metadata-filtered.tsv \
+  >   --output-strains $TMP/strains-filtered.txt \
+  >   --output-sequences $TMP/sequences-filtered.fasta \
+  >   > /dev/null 2>&1
+  $ wc -l $TMP/metadata-filtered.tsv
+  \s*5 .* (re)
+  $ wc -l $TMP/strains-filtered.txt
+  \s*4 .* (re)
+  $ wc -l $TMP/sequences-filtered.fasta
+  \s*8 .* (re)
```

Comment:
I think this test would be easier to interpret (both reading it in abstraction and when inspecting failures) if instead of relying on line counts the expected contents were actually listed out here. The files are so small it seems amenable to that. This will also avoid false positives where the line counts match but the actual content is still bogus (e.g. the wrong record is skipped because of a bug and the force-included record is still output twice).

If we do stick with line counts, I think the test here and below can be made simpler by using this construction instead:

```suggestion
  $ wc -l < $TMP/metadata-filtered.tsv
  5
  $ wc -l < $TMP/strains-filtered.txt
  4
  $ wc -l < $TMP/sequences-filtered.fasta
  8
```

Reply:
Updated to compare file contents (with sorting): 2187f78b42e1c8e251ae9e4410d334b33452b8d2

Reply:
Noting that the line count test suggestion above doesn't completely work since there is still whitespace that needs to be matched:

```
   $ wc -l < "$TMP/metadata-filtered.tsv"
-  5
+         5
```

So I think the best that can be done is:

```
  $ wc -l < $TMP/metadata-filtered.tsv
  \s*5 (re)
```

Reply:
FYI, I ran into this today while writing tests for forecasting-ncov. Seems like the difference in output is due to the difference in `wc` between Mac and Ubuntu. I was able to get around it with
```
$ wc -l < $TMP/metadata-filtered.tsv | sed 's/^[[:space:]]*//'
5
```

Reply:
Oh that works too! To me it's about the same complexity, either stripping any whitespace before output, or matching any whitespace in the output. I don't have a preference, just noting that the `\s*` is used in other Augur Cram test files.

Also nice tests organization in nextstrain/forecasts-ncov@11f50296c183fc71ff69abeb0bd49ee4bdc9ce0e!

Reply:
I learned from your [reorganization of the filter tests in augur ](https://github.com/nextstrain/augur/pull/943) 😄